l10n_br_account_payment_order/models/account_payment_line.py

```diff
@@ -96,6 +83,32 @@ class AccountPaymentLine(models.Model):
         help='Campo G048 do CNAB',
     )
 
+    payment_mode_id = fields.Many2one(
+        comodel_name='account.payment.mode',
+        string='Payment Mode',
+        ondelete='set null',
+        domain="[('id', 'in', payment_mode_ids)]",
+    )
+
+    payment_mode_ids = fields.Many2many(
```

Comment:
Qual o motivo de ter esse campo aqui ? Já que uma linha só poderá ser associada a um Payment Mode, e caso seja preciso essa informação pode ser obtida pelo campo order_id

Reply:
Por causa da usabilidade. Eu não consegui uma forma de ajustar o domínio dinamicamente pelos modos de pagamento selecionados no order_id. Quando coloquei no onchange o domínio dinâmico não funcionou, e quando coloquei "[('id', 'in', order_id.payment_mode_ids)]" deu um erro de javascript. Com o campo related readonly foi o único jeito que funcionou.

Reply:
Sim isso "[('id', 'in', order_id.payment_mode_ids)]" não funciona, minha pergunta tem relação com a criação da Ordem de Pagto somente através da Fatura por isso não havia entendido a necessidade de ter esse campo aqui, mas me parece que o que você fez tem relação com a questão do "documento de origem", isso é você precisa desse campo pois alguns casos seriam cadastrados diretamente na tela/objeto account.payment.order seria isso mesmo ? Como o @mileo escreveu isso "Pare esta prova de conceito sim, mas em um segundo momento as informações deverão vir de outros módulos do sistema, como por exemplo a fatura." seria importante já identificar esses casos adicionando dados de demo do que é feito através de uma Fatura/account.invoice é o que não é, até para serem usados no testes e confirmando se realmente existe o caso de cadastro direto na Ordem de Pagto, já que também será necessário sobre escrever o método create_account_payment_line do objeto account.invoice ( https://github.com/OCA/bank-payment/blob/12.0/account_payment_order/models/account_invoice.py#L67 ) para identificar quando será preciso criar uma nova Ordem de Pagto/account.payment.order ou incluir a linha em uma já existente no caso do Modo de Pagto relacionado estiver com o campo is_cnab_lot marcado.